lms/djangoapps/student_dashboard/test/test_views.py

```diff
@@ -0,0 +1,179 @@
+                                           course__end=datetime.utcnow() + timedelta(days=2))]
+
+    @factory.django.mute_signals(signals.pre_save, signals.post_save)
+    def _init(self, set_interests=True, add_settings=True, initialize=True, add_course_overviews=True):
+        """
+        _init method is for data set initialization.
+        """
+        if set_interests:
+            self.user.extended_profile.interests = TestStudenDashboard.interests
+            self.user.extended_profile.save()
+
+        if add_course_overviews:
+            self.course_card = CourseCardFactory(course_id=self.course.id, course_name=self.course.name)
+            self.course_overview = CourseOverviewFactory(display_name='test_overview',
+                                                         **TestStudenDashboard.course_overview_kwargs)
+
+            self.orphan_course = CourseFactory.create()
+            self.orphan_course_overview = CourseOverviewFactory(display_name='orphan_test_overview',
+                                                                id=self.orphan_course.id,
+                                                                **TestStudenDashboard.course_overview_kwargs)
+            self.orphan_course_card = CourseCardFactory(course_id=self.orphan_course.id,
+                                                        course_name=self.orphan_course.name)
+        if initialize:
+            initialize_course_settings(self.course, self.course_overview, True)
+            initialize_course_settings(self.orphan_course, self.orphan_course_overview, True)
+
+        if add_settings:
+            _settings, _created = CustomSettings.objects.get_or_create(id=self.course.id)
+            _orphan_settings, _orphan_created = CustomSettings.objects.get_or_create(id=self.orphan_course.id)
+
+            _settings.tags = \
+                _orphan_settings.tags = 'Strategy and planning|Leadership and governance|Program design and development'
+
+            _settings.save()
+            _orphan_settings.save()
+
+    @ddt.data(200, 500)
+    def test_get_joined_communities(self, status):
+        """
+        test_get_joined_communities test method is covering
+        lms.djangoapps.student_dashboard.views.get_joined_communities
+        """
+        with mock.patch.object(ForumCategory, 'joined') as joined:
+            joined.return_value = status, []
+            communities = get_joined_communities('some-username')
+            self.assertEqual(communities, [])
+
+    @ddt.data('onboarding', 'other')
+    def test_get_recommended_xmodule_courses(self, _from):
+        """
+             test_get_recommended_xmodule_courses is covering
+             lms.djangoapps.student_dashboard.views.get_recommended_xmodule_courses
+        """
+        recommended_courses = get_recommended_xmodule_courses(self.request, _from)
+        self.assertEqual(len(recommended_courses), 0)
+
+    @ddt.data('onboarding', 'other')
```

Comment:
This will run each and every test two times, So I'd suggest not do this for every test case. even in this particular case where we are testing this method with a course, without tags. We won't even reach to that point where there is a sanity on `form` so basically it will run the same test two times. Which will definitely increate the total times taken by unit tests.
Let every test run with default `form` (`onboarding`) and create a test with a form `other`. and then in that test we will assert on the values of returned course.

Reply:
@ahmed-faraz46 it is as you had suggested in reviews before, it was the point I was trying to explain it to you and might not be able to explain well. 
The suggestion you had made was to test will all possible scenarios. these are the scenarios I can cover if you want me to add more please let me know I will add more scenarios.